test/org.jkiss.dbeaver.test.platform/src/org/jkiss/dbeaver/model/sql/analyzer/SQLCompletionAnalyzerTest.java

```diff
@@ -147,6 +147,26 @@ public void testColumnNamesCompletion() throws DBException {
         }
     }
 
+    @Test
+    public void testColumnWithNonExistingAliases() throws DBException {
+        final RequestResult request = RequestBuilder.tables(s-> {
+            s.table("Table1", t -> {
+                t.attribute("Col1");
+                t.attribute("Col2");
+            });
+            s.table("Table2", t -> {
+                t.attribute("Col4");
+                t.attribute("Col5");
+            });
+        }).prepare();
+            {
+                final List<SQLCompletionProposalBase> proposals = request.request("SELECT * FROM Table1 join Table2 t on t.|", false);
+                Assert.assertEquals("Col4", proposals.get(0).getReplacementString());
+                Assert.assertEquals("Col5", proposals.get(1).getReplacementString());
+            }
+
```

Comment:
Please fix indentation and remove excess newline at the end of this selection

Reply:
Also, I'd like to see an additional test for `SELECT * FROM Table1 t join Table2 on t.|`.